tests/snuba/api/endpoints/test_organization_events_meta.py

```diff
@@ -390,3 +390,38 @@ def test_related_issues_transactions_with_quotes(self):
         assert len(response.data) == 1
         assert response.data[0]["shortId"] == event.group.qualified_short_id
         assert int(response.data[0]["id"]) == event.group_id
+
+
+class OrganizationSpansSamplesEndpoint(APITestCase, SnubaTestCase):
+    url_name = "sentry-api-0-organization-spans-samples"
+
+    @mock.patch(
+        "sentry.search.events.builder.spans_indexed.SpansIndexedQueryBuilder.process_results",
+        return_value={"data": []},
+    )
+    def test_is_segment_properly_converted_in_filter(self, mock_request):
+        self.login_as(user=self.user)
+        project = self.create_project()
+        url = reverse(self.url_name, kwargs={"organization_slug": project.organization.slug})
+
+        response = self.client.get(
+            url,
+            {
+                "query": "span.is_segment:1 transaction:api/0/foo",
+                "lowerBound": 0,
+                "firstBound": 10,
+                "secondBound": 20,
+                "upperBound": 200,
+                "column": "span.duration",
+            },
+            format="json",
+            extra={"project": [project.id]},
+        )
+
+        assert response.status_code == 200, response.content
+
+        # the SQL should have is_segment converted into an int for all requests
+        assert all(
+            "equals((is_segment AS _snuba_is_segment), 1)" in call_args[0][0]["sql"]
+            for call_args in mock_request.call_args_list
+        )
```

Comment:
@shruthilayaj I'm not exactly sure how I was supposed to test the SQL, do we have any other examples where we do that? This feels pretty brittle because it's not testing where in the SQL query this condition is

Also, you mentioned testing the downstream effects of this but I ended up deciding that it didn't feel that useful since we don't have the rest of the infrastructure in place to test this. If I wanted to mock the results of fetching the span IDs, then there's another query downstream to fetch span data which I'd also have to mock